app/controllers/v1/resources_controller.rb

```diff
@@ -0,0 +1,304 @@
+# frozen_string_literal: true
+
+require 'base64'
+
+module V1
+  # rubocop:disable Metrics/ClassLength
+  class ResourcesController < ApplicationController
+    include Authenticated
+
+    class BlobError < StandardError; end
+    class GlobusNotFoundError < StandardError; end
+
+    before_action :check_auth_token, only: %i[create update]
+    before_action :validate_version, only: %i[create update]
+
+    GLOBUS_PREFIX = 'globus://'
+
+    # GET /resource/:id
+    def show
+      cocina_obj = Cocina::Models.without_metadata(Dor::Services::Client.object(params[:id]).find)
+      authorize! cocina_obj, with: ResourcePolicy
+      render json: cocina_obj
+    rescue Dor::Services::Client::NotFoundResponse => e
+      render build_error('404', e, "Object not found: #{params[:id]}")
+    rescue Dor::Services::Client::UnexpectedResponse => e
+      render build_error('500', e, 'Internal server error')
+    end
+
+    # POST /resource
+    # rubocop:disable Metrics/MethodLength
+    # rubocop:disable Metrics/AbcSize
+    def create
+      begin
+        @druid = cocina_object.externalIdentifier
+        shelve_files
+        location = write_purl
```

Comment:
You must also stick this stuff in here:
https://github.com/sul-dlss/purl-fetcher/blob/main/app/controllers/v1/purls_controller.rb#L16-L22